### PR TITLE
Add 'out' function to output arbitrary data from iterator.

### PR DESCRIPTION
**Pull request by nizomovs**
This PR adds a new `out` function with a new function type parameter. The function gets as input the current value that is iterated over and can return arbitrary data which is then written to the output stream.

### CODE REVIEW

File: include/gdsb/experiment.h [hunk 1]
```diff
@@ -68,6 +68,22 @@ template <typename It> void out(std::string name, It begin, It end, std::ostream
...
+//! Use this function to output ranges in yaml format.
+template <typename It, typename F>
+void out(
+    std::string name, It begin, It end, std::ostream& stream = std::cout, F&& access = [](auto const& e) { return e; })
+{
+    stream << name << ": [";
+
+    if (begin != end)
+    {
+        std::for_each(begin, end - 1, [&](auto const& e) { stream << access(e) << ", "; });
```
Comment (c-bebop) [inline include/gdsb/experiment.h:80]: Can you write a test for this function? The test for `out()` using iterators can be found here: https://github.com/hu-macsy/graph-ds-benchmark/blob/main/test/experiment_tests.cpp#L31

File: include/gdsb/experiment.h [hunk 2]
```diff
@@ -68,6 +68,22 @@ template <typename It> void out(std::string name, It begin, It end, std::ostream
     stream << "]" << std::endl;
 }
 
+//! Use this function to output ranges in yaml format.
+template <typename It, typename F>
+void out(
+    std::string name, It begin, It end, std::ostream& stream = std::cout, F&& access = [](auto const& e) { return e; })
```
[the remark this answers is not in the document]
Comment (nizomovs) [reply include/gdsb/experiment.h]: I have changed the order of the function parameters. Additionally, I had to change the `access` parameter type to `F const&&`.

File: include/gdsb/experiment.h (hunk 1 above)
Comment (nizomovs) [reply include/gdsb/experiment.h:79]: There's now a test similar to the other tests.

File: include/gdsb/experiment.h [hunk 3]
```diff
@@ -68,6 +68,21 @@ template <typename It> void out(std::string name, It begin, It end, std::ostream
...
 
+//! Use this function to output ranges in yaml format.
+template <typename It, typename F>
+void out(std::string name, It begin, It end, F const&& access, std::ostream& stream = std::cout)
+{
+    stream << name << ": [";
+
+    if (begin != end)
+    {
+        std::for_each(begin, end - 1, [&](auto const& e) { stream << access(e) << ", "; });
```
Comment (c-bebop) [inline include/gdsb/experiment.h:79]: Is it necessary to capture `&`? Why not capture `access` as `access = std::move(acces)`?

**Review by c-bebop — CHANGES_REQUESTED** [review]

**Review by c-bebop — APPROVED** [review]

**Pull request merged**